src/inputs/run-install.ts

```diff
@@ -1,70 +1,50 @@
-import { getInput, InputOptions, error } from '@actions/core'
-import { parse } from 'yaml'
+import { getInput, error } from '@actions/core'
+import * as yaml from 'yaml'
+import { z, ZodError } from 'zod'
 
 export interface RunInstall {
   readonly recursive?: boolean
   readonly cwd?: string
   readonly args?: readonly string[]
 }
 
+const zRunInstall = z.object({
+  recursive: z.boolean().optional(),
+  cwd: z.string().optional(),
+  args: z.array(z.string()).optional(),
+})
+
```

Comment:
The `z` prefix is distracting. TypeScript allows name collision between runtime identifier and type identifier, so you can remove it. Also use `typeof` to avoid duplication.

```suggestion
const RunInstall = z.object({
  recursive: z.boolean().optional(),
  cwd: z.string().optional(),
  args: z.array(z.string()).optional(),
})

export type RunInstall = typeof RunInstall
```

Reply:
Found [`z.infer`](https://github.com/colinhacks/zod?tab=readme-ov-file#guides-and-concepts) which resolves the types. Otherwise they look like:

```
z.ZodUnion<[z.ZodNull, z.ZodBoolean, z.ZodObject<{
    recursive: z.ZodOptional<z.ZodBoolean>;
    cwd: z.ZodOptional<z.ZodString>;
    args: z.ZodOptional<z.ZodArray<z.ZodString, "many">>;
}, "strip", z.ZodTypeAny, {
    ...;
}, {
    ...;
}>, z.ZodArray<...>]>
```